src/cloud-api-adaptor/docs/SecureComms.md

```diff
@@ -0,0 +1,125 @@
+```sh
+apiVersion: v1
+data:
+    ...
+    SECURE_COMMS: "true"
+    ...
+```
+
+You may also include additional Inbounds and Outbounds configurations to the Adaptor using the `SECURE_COMMS_INBOUNDS` and `SECURE_COMMS_OUTBOUNDS` config points. See more details regarding Inbounds and Outbounds below.
+
+You may also set the KBS address using the `SECURE_COMMS_KBS_ADDR` config point.
+
+
+### Adding named tunnels to the SSH channel
+Named tunnels can be added to the SSH channel. Adding a named tunnel requires adding an Inbound at one of the SSH channel peers and an Outbound at the other SSH channel peer. The Inbound and Outbound both carry the name of the tunnel being created.
+
+        |---------Tunnel----------| 
+Client->Inbound----------->Outbound->Server
+
+
+Inbounds and Outbounds take the form of a comma separated inbound/outbound tags such that Inbounds are formed as "InboundTag1,InboundTag2,InboundTag3,..." and Outbounds are formed as "OutboundTag1,OutboundTag2,outboundTag3,..."
+
+Each Inbound tag is structured as `Phase:Name:Port` where:
+- Phase can be 'K' to represent an inbound available during the Kubernetes phase, 'A' to represent an inbound available during the Attestation phase, or 'B' to represent an inbound available during both phases.
+- Name is the name of the tunnel
+- Port is the local service port being opened to serve as ingress of the tunnel.
+
+Each outbound tag is structured as `Phase:Name:Host:Port` or `Phase:Name:Port` where:
+- Phase can be 'KUBERNETES_PHASE' to represent an outbound available during the Kubernetes phase, 'ATTESTATION_PHASE' to represent an outbound available during the Attestation phase, or 'BOTH_PHASES' to represent an outbound available during both phases.
+- Name is the name of the tunnel
+- Host (if available) is a hostname or IP address of the destination host to forward the information to.
+- Port is the destination port to forward the information to.
+
+For example, an outbound tag such as `KUBERNETES_PHASE:ABC:myhost.com:1234` means that during the `Kubernetes phase`, an output of a tunnel named `ABC` is registered, such that information from a client connecting to ABC Inbound will be tunneled and forwarded to `myhost.com` port `1234`).
+
+## Testing
+
+Testing securecomms as a standalone can be done by using:
+```sh
+cd src/cloud-api-adaptor
+go run ./test/securecomms/double/main.go
+```
+This will create a client and a server and mimic the connection between a CAA client to a PP Server
+Successful connection result in exit code of 0
+
+Alternatively, the client and server can be separately executed in independent terminals using `./test/securecomms/double/wnssh.go` and `./test/securecomms/ppssh/main.go` respectively.
+
+## Future Plans
+
+- Add DeleteResource() support in KBS, KBC, api-server-rest, than cleanup resources added by Secure Comms to KBS whenever a Peer Pod fail to be created or when a Peer Pod is terminated.
```

Comment:
I would like to see Pradipta's suggestion of moving the agent-protocol-forwarder config to a file and provide it via user-data which process-user-data can read and create the config, such that we don't need a separate podvm build to use this feature, on the future plans list as that would help enable much easier testing & release of this feature.

Reply:
Since this doc is specific for the secure comms and using a config file for agent-protocol-forwarder is not really specific to secure comms, my suggestion will be to create a tracker issue in this repo.

Reply:
That works for me too